arch/arm/core/cortex_m/mpu/Kconfig

```diff
@@ -38,3 +38,10 @@ config NXP_MPU
 	default n
 	help
 	  MCU has NXP MPU
+
+config MPU_ALLOW_FLASH_WRITE
+	bool "Add MPU access to write to flash"
+	depends on NXP_MPU
+	default n
```

Comment:
Thanks a lot for submitting this! It will really help us.

What do people think about making this option `default y if FLASH` or so?

That seems less surprising; by default, asking for both the flash driver and the MPU driver would result in a config with fewer pointy edges. User effort would only be needed to get the flash driver and to allow flash reads, but not writes. @galak pointed out that this is a confusing configuration whose utility is a bit unclear, so perhaps that setup should take more effort to create.